### ISSUE TITLE
Vaulting & Pay Later UI/UX

### ISSUE THREAD

**Issue by jorgeatorres** [woocommerce/woocommerce-paypal-payments#174]
Progress has been made since #157, but I believe there are still some aspects that need fine-tuning and it'd be great if we could start discussing approaches or ideas.

1. Enabling vaulting is still cumbersome as you have to uncheck each and every _Pay Later_ checkbox one by one. This is not very user-friendly, especially considering that _Pay Later_ comes enabled by default. It might seem like a minor detail, but there's no global _Pay Later_ switch and the settings are scattered around the settings page, making things unnecessarily difficult.

   I'm no longer convinced the approach I suggested in #157 (displaying a confirmation dialog and doing the unchecking automatically with JS) is the best one, but I still think we should make things easier. Maybe the vaulting setting should only be disabled when the token doesn't have the necessary scope, but in every other case it should be possible to enable vaulting (without asking for confirmation) as long as it's clear that enabling it will disable Pay Later. The setting could look something like this:

   <img width="694" alt="Screen Shot 2021-06-10 at 08 07 08" src="https://user-images.githubusercontent.com/184724/121530689-4813e080-c9c3-11eb-9267-281f73d3c208.png">

   Obviously the "warning" would only appear when vaulting hasn't been enabled.

2. It's not immediately clear _why_ the vaulting setting might be disabled. This notice at the top does explain why but it is easy to miss. It also seems a bit superfluous given _Pay Later_ is enabled by default, so it's a notice that almost every merchant is getting even if they don't need or want to use vaulting:
  
   <img width="1251" alt="Screen Shot 2021-06-10 at 07 55 13" src="https://user-images.githubusercontent.com/184724/121528937-88725f00-c9c1-11eb-9afe-cd0a7bf1a41b.png">

   We should probably move the "explanation" next to the vaulting checkbox itself, so you only see it when you actually want to enable vaulting:
   
   <img width="1133" alt="Screen Shot 2021-06-10 at 07 54 52" src="https://user-images.githubusercontent.com/184724/121529152-bbb4ee00-c9c1-11eb-8f27-71a310ce4562.png">

   To make things worse, the disabled state plus the link to enable vaulting give the idea that the setting is disabled because you haven't done something with your PayPal account and not because some _Pay Later_ checkbox is checked off below.

   That said, if we go with what I suggested in (1), it wouldn't be necessary to disable the vaulting checkbox at all, except in the specific case of the token not containing the required scopes. Only in that particular case, we would display the link to "enable vaulting on your account".

3. _Pay Later_ settings are better at indicating why you can't use them, but the messaging remains even when you've disabled vaulting:

   ![screencast](https://user-images.githubusercontent.com/184724/121531849-675f3d80-c9c4-11eb-83ec-ddac6581c402.gif)

   The messaging should reflect the current configuration (even if unsaved) and not be tied to the values in the database. Also, the warning should have an anchor link to the vaulting setting so that you can easily toggle it.

To summarize my thoughts: (I) as long as the scope is in the token, it should always be possible to enable vaulting (with a small warning indicating that doing so will disable _Pay Later_ messaging) (II) the UI in general (labels, warnings, etc.) should always reflect the current configuration as seen by the user (even if it hasn't been saved to the database).

Please let me know what you think.
